nucypher/blockchain/eth/signers.py

```diff
@@ -191,3 +216,93 @@ def unlock_account(self, account: str, password: str, duration: int = None) -> b
     @validate_checksum_address
     def lock_account(self, account: str):
         return True
+
+
+class KeyStoreSigner(Signer):
+
+    IS_SIGNER_LOCAL = True
+
+    def __init__(self, uri: str):
+        super().__init__()
+
+        decoded_uri = urlparse(uri)
+        if decoded_uri.scheme not in ('key', 'keystore') or decoded_uri.netloc:
+            raise cls.InvalidSignerURI(uri)
+
+        with open(decoded_uri.path) as file_obj:
+            key = json.load(file_obj)
+
+        address = key['address']
+        if not Web3.isAddress(address):
+            raise ValueError(f"'{decoded_uri.path}' does not contain a valid address")
+
+        self.key = key
+        self._account = Web3.toChecksumAddress(address)
+        self._signer = None
+
+    def _is_my_account(self, account) -> bool:
+        return Web3.isAddress(account) and \
+            self.account == Web3.toChecksumAddress(account)
+
+    @classmethod
+    def from_signer_uri(cls, uri: str) -> 'Signer':
+        return cls(uri)
+
+    @property
+    def account(self) -> str:
+        return self._account
+
+    @validate_checksum_address
+    def is_device(self, account: str) -> bool:
+        return False
+
+    def accounts(self) -> List[str]:
+        return [self.account]
+
+    @validate_checksum_address
+    def unlock_account(self, account: str, password: str, duration: int = None) -> bytes:
+        if not self._is_my_account(account):
+            return False
+
+        if self._signer is None:
+            try:
+                self._signer = Account.from_key(
+                    Account.decrypt(self.key, password)
+                )
+            except ValueError:
+                return False
+
+            if self.account != self._signer.address:
+                raise ValueError("unexpected address '%s' recovered from private key; expected '%s'" % (
+                    self._signer.address,
+                    self.account,
+                ))
+
+        return True
+
+    @validate_checksum_address
+    def lock_account(self, account: str) -> str:
+        if not self._is_my_account(account):
+            return False
+
+        self._signer = None
+        return True
+
+    @validate_checksum_address
+    def sign_transaction(self, transaction_dict: dict) -> HexBytes:
+        # Do not include a 'to' field for contract creation.
+        if not transaction_dict['to']:
+            transaction_dict = dissoc(transaction_dict, 'to')
+
```

Comment:
This method needs to have a sanity check to verify that the 'from' field matches the local account...  Is that included in `Account.sign_transaction`?

Reply:
Yes, [here](https://github.com/ethereum/eth-account/blob/82d1205e5d1ba591734356cf65b7c0c16eade579/eth_account/account.py#L638)

Reply:
Yes - good.  Do we want the `TypeError` to bubble up here?

Reply:
I should think so - I believe you would get an RPC error in this situation with Web3Signer.